### PR TITLE
Fix the OpenID Connect module

### PR DESCRIPTION
**Pull request by PinpointTownes**
@sebastienros please review when you have a minute :smile: 

/cc @jersiovic

### CODE REVIEW

File: src/Orchard.Web/Modules/Orchard.OpenId/Startup.cs [hunk 1]
```diff
@@ -27,13 +28,13 @@ public class Startup : StartupBase
 
         public Startup(
             ShellSettings shellSettings,
-            IAppDataFolder appDataFolder,
+            IHostingEnvironment environment,
```
Comment (PinpointTownes) [inline src/Orchard.Web/Modules/Orchard.OpenId/Startup.cs]: @sebastienros not sure this is the right fix, so please correct me if I'm wrong :smile:

File: src/Orchard.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs [hunk 2]
```diff
@@ -4,18 +4,19 @@
...
+using OpenIddict.Mvc;
+using Orchard.Mvc;
 using Orchard.OpenId.Services;
 using Orchard.OpenId.ViewModels;
 using Orchard.Users.Models;
 
 namespace Orchard.OpenId.Controllers
 {
-    [Authorize, ValidateAntiForgeryToken]
+    [Authorize, AutoValidateAntiforgeryToken]
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs]: Isn't it supposed to be applied at global level only, which is already done? Or is that an extraneous safety level to ensure it's really there. But then the previous one would be fine too.

https://github.com/aspnet/Announcements/issues/156

File: src/Orchard.Web/Modules/Orchard.OpenId/Startup.cs (hunk 1 above)
Comment (Jetski5822) [inline src/Orchard.Web/Modules/Orchard.OpenId/Startup.cs]: Inject 

IHostingEnvironment environment,
IOptions<ShellOptions> optionsAccessor

then 

Path.Combine(
          environment.ContentRootPath,  // orchard.web/
          optionsAccessor.Value.ShellsRootContainerName, // app_data 
          optionsAccessor.Value.ShellsContainerName, // sites
          shellSettings.Name,  // sitename
          CertificateFileName);

File: src/Orchard.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs (hunk 2 above)
Comment (PinpointTownes) [inline src/Orchard.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs]: > Or is that an extraneous safety level to ensure it's really there.

Exactly that. I'm paranoid :smile: 
If you think it's not worth it, I'm fine removing it.

File: src/Orchard.Web/Modules/Orchard.OpenId/Startup.cs (hunk 1 above)
Comment (PinpointTownes) [inline src/Orchard.Web/Modules/Orchard.OpenId/Startup.cs]: Okay, looks like `ShellsRootContainerName` and `ShellsContainerName` were introduced yesterday just after I "rebased" the `oauth2` branch. I'll fix that, thanks!

File: src/Orchard.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs (hunk 2 above)
Comment (PinpointTownes) [inline src/Orchard.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs]: Qui ne dit mot consent :smile:

**Pull request merged**